File: 架构模式-MVP/架构模式-MVP/MVP/Presenter/CategoryPresenter.m
//
//  CategoryPresenter.m
//  架构模式-MVP
//
//  Created by 陈良静 on 2017/4/27.
//  Copyright © 2017年 陈良静. All rights reserved.
//

#import "CategoryPresenter.h"
#import "CategoryModel.h"

@interface CategoryPresenter ()

@property(nonatomic, strong) CategoryModel *categoryM;
@property(nonatomic, weak) id<CategroyTableViewDelegate> categoryT;

@end

@implementation CategoryPresenter

- (instancetype)init
{
    self = [super init];
    if (self) {
        _categoryM = [[CategoryModel alloc] init];
    }
    
    return self;
}

// 绑定view
- (void)attachView:(id<CategroyTableViewDelegate>)categoryTableView
{
    _categoryT = categoryTableView;
}

// 解除绑定View
- (void)detachView
{
    _categoryT = nil;
}

- (void)categoryTableViewRequestDataWithName:(NSString *)name pwd:(NSString *)pwd
{
    [_categoryM requestCategoryDataWithName:name pwd:pwd callBack:^(id response) {
        if (_categoryT != nil && [_categoryT respondsToSelector:@selector(onCategroyTableViewResult:)])
        {
            [_categoryT onCategroyTableViewResult:response];
        }
    }];
}

@end

File: iOS架构模式—MVP（Swift）/iOS架构模式—MVP（Swift）/CategoryPresenter.swift
//
//  CategoryPresenter.swift
//  iOS架构模式—MVP（Swift）
//
//  Created by 陈良静 on 2017/5/1.
//  Copyright © 2017年 陈良静. All rights reserved.
//

import Foundation

var categoryM : Category?

var categoryT : AnyClass?


func attchView(_ view: AnyClass) -> Void {
    categoryT = view
}

func detachView() {
    categoryT = nil
}

func categoryTableViewRequsetData(WithName name : String, pwd : String) -> Void  {
    categoryM?.requsetCategoryData(WithName: name, pwd: pwd, callBack: {(_ resonse : Any) in
        
    }
}
